back/src/server.ts

```diff
@@ -37,25 +43,26 @@ const {
 
 const UI_BASE_URL = getUIBaseURL();
 
-const typesArray = fileLoader(`${__dirname}/**/*.graphql`, { recursive: true });
-const typeDefs = mergeTypes(typesArray, { all: true });
+const shieldMiddleware = shield(permissions, { allowExternalErrors: true });
 
-const resolversArray = fileLoader(`${__dirname}/**/resolvers.ts`, {
-  recursive: true
-});
-const resolvers = mergeResolvers(resolversArray);
-
-const permissions = fileLoader(`${__dirname}/**/permissions.ts`, {
-  recursive: true
-});
-const shieldMiddleware = shield(mergePermissions(permissions));
+const schemaValidationMiddleware = schemaValidation(validations);
 
-const schemas = fileLoader(`${__dirname}/**/schema-validation.ts`, {
-  recursive: true
-});
-const schemaValidationMiddleware = schemaValidation(
-  mergeValidationRules(schemas)
-);
+/**
+ * Custom report error for sentry middleware
+ * It decides whether or not the error should be captured
+ */
+export function reportError(res: Error | any) {
+  if (
+    res instanceof UserInputError ||
+    res instanceof AuthenticationError ||
+    res instanceof ForbiddenError ||
+    res instanceof SyntaxError ||
+    res instanceof ValidationError
```

Comment:
Tout [extend `ApolloError`](https://github.com/apollographql/apollo-server/blob/faba52c689c22472a19fcb65d78925df549077f7/packages/apollo-server-errors/src/index.ts#L196) je crois. On pourrait peut etre remplacer par `instanceof ApolloError` ?
Ca permet egalement de throw manuellement des `ApolloError`

Reply:
C'est très bizarre, j'ai l'impression que le type de l'erreur n'est pas préservé dans la fonction `reportError`. Si je throw une `UserInputError` dans le resolvers, `res instanceOf UserInputError` est à `false`. Je suis obligé de faire un match sur le code erreur

Reply:
Peut-être lié à https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
Just a guess, mais ce n'est pas une `GraphQLError` avec une property `originalError` qui a le bon type ?

Reply:
Dans le middleware, l'erreur n'a pas encore été transformé en `GraphQLError`, `originalError` est undefined

Reply:
@providenz le test suivant passe donc je ne pense pas que ce soit ça

```
  test("error types", () => {
    expect(new UserInputError("Bang") instanceof UserInputError).toBeTruthy();
    expect(new UserInputError("Bang") instanceof ApolloError).toBeTruthy();
  });
```

Reply:
Bon je comprend pas trop... du coup je fais le match sur le code d'extension, ça me parait plus fiable.